core/kernel/user_access.c

```diff
@@ -17,9 +17,26 @@
 static TEE_Result check_access(uint32_t flags, const void *uaddr, size_t len)
 {
 	struct ts_session *s = ts_get_current_session();
+	struct user_mode_ctx *uctx = NULL;
+	
+	if (!is_user_mode_ctx(s->ctx)) {
+		/*
+		 * check_access() may be called within a PTA session, which
+		 * doesn't have a user_mode_ctx. Here, try to retrieve the
+		 * user_mode_ctx associated with the calling session.
+		 */
+		s = TAILQ_NEXT(s, link_tsd);
+		if (!s)
+			return TEE_ERROR_ACCESS_DENIED;
+
+		uctx = to_user_mode_ctx(s->ctx);
+		if (!uctx)
+			return TEE_ERROR_ACCESS_DENIED;
```

Comment:
should be TEE_ERROR_GENERIC as this is not expected. even `panic()` seems appropriate to me.